ldoc/lang.lua

```diff
@@ -111,9 +111,10 @@ end
 -- that we must infer the module name.
 function Lua:find_module(tok,t,v)
    local res
-   res,t,v = self:search_for_token(tok,'iden','module',t,v)
+   local st
+   res,st,v = self:search_for_token(tok,'iden','module',t,v)
```

Comment:
Thanks for tracking this down. I think you've diagnosed it correctly, I do have a small concern this fix will actually clobber `v`. Lets lay this like does not return anything for `res`, and also returns `nil` or whatever in the third position. Unless it *always* returns the same `v` as was passed to it, this is going to mutate `v` whether it fails or succeeds. That may lead to passing the wrong `v` later. Shouldn't this be `_v` or same other more explicit variable and then return `v` if `not res` but pass `_v` to `:parse_module_call()` otherwise?

Reply:
This again goes back to the call to `tnext()` in `search_for_token()`, and I can't really tell if that may clobber `v` -- so I agree, it would make sense to rename it and only "apply" the overriding of that value if its not part of an invalid result.

Reply:
I decided to do the same with `v` where I create a temporary called `sv` (`s` due to `search_...`). Doesn't seem to break anything, and probably was a bug as you said.

Reply:
That looks better. It might not have ever come up in practice but at least in theory I think that would have been a bug.